lib/src/embedded/dispatcher.ts

```diff
@@ -77,24 +74,43 @@ export class Dispatcher {
 
   constructor(
     private readonly outboundMessages$: Observable<OutboundTypedMessage>,
-    private readonly writeInboundMessage: (message: InboundTypedMessage) => void
+    private readonly writeInboundMessage: (
+      message: InboundTypedMessage
+    ) => void,
+    private readonly handleImportRequest: (
+      request: OutboundMessage.ImportRequest
+    ) => PromiseOr<InboundMessage.ImportResponse>,
+    private readonly handleFileImportRequest: (
+      request: OutboundMessage.FileImportRequest
+    ) => PromiseOr<InboundMessage.FileImportResponse>,
+    private readonly handleCanonicalizeRequest: (
+      request: OutboundMessage.CanonicalizeRequest
+    ) => PromiseOr<InboundMessage.CanonicalizeResponse>,
+    private readonly handleFunctionCallRequest: (
+      request: OutboundMessage.FunctionCallRequest
+    ) => PromiseOr<InboundMessage.FunctionCallResponse>
   ) {
-    this.outboundMessages$
-      .pipe(tap(message => this.registerOutboundMessage(message)))
-      .subscribe(
-        message => this.messages$.next(message),
-        error => this.throwAndClose(error),
-        () => {
-          this.messages$.complete();
-          this.errorInternal$.complete();
+    this.outboundMessages$.subscribe(
+      async message => {
+        try {
+          await this.handleOutboundMessage(message);
+          this.messages$.next(message);
+        } catch (error) {
+          this.throwAndClose(error);
         }
-      );
+      },
```

Comment:
Why did this go back to a `try`/`catch`? If it's so that `handleOutboundMessage` can be async, I think you can just use `flatMap()` instead of `tap()` to inject the promise errors into the result observable.

Reply:
Yeah, this was for the async. Thanks for the catch